src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,381 @@
+        }
+
+        arr[this.size - 1] = student;
+        this.array = arr;
         return false;
     }
 
     @Override
     public boolean remove(Object o) {
-        //TODO
-        return false;
+        boolean flag = false;
+        if (o != null) {
+            for(int i = 0; i < this.size; ++i) {
+                if (o.equals(this.array[i])) {
+                    flag = true;
+                }
+
+                if (flag) {
+                    this.array[i] = this.array[i + 1];
+                }
+            }
+        }
+
+        --this.size;
+        return flag;
     }
 
     @Override
     public void clear() {
-        //TODO
+        for(int i = 0; i < this.size; ++i) {
+            this.array[i] = null;
+        }
+
+        this.size = 0;
     }
 
     @Override
     public Student get(int i) {
-        //TODO
-        return null;
+        if (i > this.size && i < 0) {
+            throw new IndexOutOfBoundsException();
+        } else {
+            return this.array[i];
+        }
     }
 
     @Override
     public Student set(int i, Student student) {
-        //TODO
-        return null;
+        if (i >= 0 && i < this.size) {
+            Student previous = this.array[i];
+            this.array[i] = student;
+            return previous;
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
-    public void add(int i, Student student) {
-        //TODO
+    public void add(int index, Student student) {
+        if (index >= 0 && index < this.size) {
+            for(int i = this.size; i > index; --i) {
+                this.array[i] = this.array[i - 1];
+            }
+
+            this.array[index] = student;
+            ++this.size;
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
-    public Student remove(int i) {
-        //TODO
-        return null;
+    public Student remove(int index) {
+        if (index >= 0 && index < this.size) {
+            Student previous = this.array[index];
+
+            for(int i = index; i < this.size; ++i) {
+                this.array[i] = this.array[i + 1];
+            }
+
+            --this.size;
+            return previous;
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
     public int indexOf(Object o) {
-        //TODO
-        return 0;
+        int index = -1;
+        if (this.contains(o)) {
+            for(int i = 0; i < this.size; ++i) {
+                if (o.equals(this.array[i])) {
+                    index = i;
+                    break;
+                }
+            }
+        }
+
+        return index;
     }
 
     @Override
     public int lastIndexOf(Object o) {
-        //TODO
-        return 0;
+        int index = -1;
+        if (this.contains(o)) {
+            for(int i = 0; i < this.size; ++i) {
+                if (o.equals(this.array[i])) {
+                    index = i;
```

Comment:
you can return here to decrease complexity of the loop

Reply:
Fixed!